File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {} #dictionary to hold value and index pair
        
        for i in range(len(nums)): ##iterates through nums
            complement = target - nums[i] #complement of i
            
            if complement in visited: #checks if complement of i exists in dictionary of previously visited values
                return [visited[complement], i] #if it is, return the index of the complement and current index
            else:
                visited[nums[i]] = i #if not add current index to i
            
            
        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let a= [];
    let x; 
    
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1 ; j < nums.length; j++) {
            x = target - nums[i]
            if (nums[j] === x ){
                a = [i,j];
                return a;
            }
        }
            
    }
    
    return null;
};